homeassistant/components/devialet/media_player.py

```diff
@@ -0,0 +1,274 @@
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Devialet entry."""
+    client = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [DevialetDevice(client, entry)],
+        update_before_add=True,
+    )
+
+
+class DevialetDevice(MediaPlayerEntity):
+    """Representation of a Devialet device."""
+
+    def __init__(self, client: DevialetApi, entry: ConfigEntry) -> None:
+        """Initialize the Devialet device."""
+        self._client = client
+        self._name = entry.data[CONF_NAME]
+        self._muted = False
+        if entry.unique_id:
+            self._serial = entry.unique_id
+
+    async def async_update(self) -> None:
+        """Get the latest details from the device."""
+        await self._client.async_update()
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self._serial)},
+            name=self._name,
+            manufacturer=MANUFACTURER,
+            model=self._client.model,
+            sw_version=self._client.version,
+        )
+
+    @property
+    def name(self) -> str:
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def unique_id(self) -> str:
+        """Return the unique id of the device."""
+        return self._serial
+
+    @property
+    def state(self) -> MediaPlayerState | None:
+        """Return the state of the device."""
+        if not self._client.is_available:
+            return MediaPlayerState.OFF
+
+        playing_state = self._client.playing_state
+
+        if not playing_state:
+            return MediaPlayerState.IDLE
+        if playing_state == "playing":
+            return MediaPlayerState.PLAYING
+        if playing_state == "paused":
+            return MediaPlayerState.PAUSED
+        return MediaPlayerState.ON
+
+    @property
+    def volume_level(self) -> float | None:
+        """Volume level of the media player (0..1)."""
+        return self._client.volume_level
+
+    @property
+    def is_volume_muted(self) -> bool | None:
+        """Return boolean if volume is currently muted."""
+        return self._client.is_volume_muted
+
+    @property
+    def source_list(self) -> list[str] | None:
+        """Return the list of available input sources."""
+        return self._client.source_list
+
+    @property
+    def sound_mode_list(self) -> list[str] | None:
+        """Return the list of available sound modes."""
+        return sorted(SOUND_MODES)
+
+    @property
+    def media_artist(self) -> str | None:
+        """Artist of current playing media, music track only."""
+        return self._client.media_artist
+
+    @property
+    def media_album_name(self) -> str | None:
+        """Album name of current playing media, music track only."""
+        return self._client.media_album_name
```

Comment:
These properties can all be converted to the shorthand instance attributes

Reply:
For non-changing attributes (like ICON), yes. But in this case it cannot be set in the init, because it requires a call to the device first.

Reply:
It doesn't need to happen in __init__, it can happen in update:
```py
    async def async_update(self) -> None:
        """Get the latest details from the device."""
        await self._client.async_update()
        self._attr_media_album_name = self._client.media_album_name
        ...
```

This reduces the line count a great deal, which is a good thing